deploy-manage/upgrade/deployment-or-cluster.md

```diff
@@ -59,16 +54,366 @@ $$$upgrading-reindex$$$
+% * [/raw-migrated-files/cloud/cloud/ec-upgrade-deployment.md](/raw-migrated-files/cloud/cloud/ec-upgrade-deployment.md)
+% * [/raw-migrated-files/stack-docs/elastic-stack/upgrade-elastic-stack-for-elastic-cloud.md](/raw-migrated-files/stack-docs/elastic-stack/upgrade-elastic-stack-for-elastic-cloud.md)
+% * [/raw-migrated-files/stack-docs/elastic-stack/upgrading-elastic-stack-on-prem.md](/raw-migrated-files/stack-docs/elastic-stack/upgrading-elastic-stack-on-prem.md)
+% * [/raw-migrated-files/cloud-on-k8s/cloud-on-k8s/k8s-upgrading-stack.md](/raw-migrated-files/cloud-on-k8s/cloud-on-k8s/k8s-upgrading-stack.md)
+% * [/raw-migrated-files/cloud-on-k8s/cloud-on-k8s/k8s-orchestration.md](/raw-migrated-files/cloud-on-k8s/cloud-on-k8s/k8s-orchestration.md)
+
+# Upgrade your deployment or cluster [upgrade-deployment-cluster]
+
+When upgrading the version of an existing cluster, you perform either a minor or major upgrade. The difference is that a minor upgrade takes you from version 9.0 to 9.1, for example, while a major upgrade takes you from version 8 to 9. 
+
+The procedures you follow to upgrade depend on whether you’ve installed Elastic components using Elastic-managed infrastructure or self-managed infrastructure.
+
+If you’re running Elastic-managed infrastructure, your options are to:
+
+* [Upgrade on {{ech}}](/deploy-manage/upgrade/deployment-or-cluster/upgrade-on-ech.md)
+* Upgrade on {{serverless-full}} (updates are automatic and require no user management)
+
+If you’re running your own self-managed infrastructure — either on-prem or on public cloud — your options are to:
+
+* [Upgrade the {{stack}}](/deploy-manage/upgrade/deployment-or-cluster/self-managed.md) (upgrade each component individually)
+* [Upgrade on {{ece}} (ECE)](/deploy-manage/upgrade/deployment-or-cluster/upgrade-on-ece.md)
+* [Upgrade on {{eck}} (ECK)](/deploy-manage/upgrade/deployment-or-cluster/upgrade-on-eck.md)
+
+## Prepare to upgrade [prepare-to-upgrade]
+
+Before you upgrade Elastic, it's important to take some preparation steps. These steps vary based on your current version. 
+
+:::{important}
+Upgrading from a release candidate build, such as 9.0.0-rc1 or 9.0.0-rc2, is not supported. Pre-releases should only be used for testing in a temporary environment.
+:::
+
+## Prepare to upgrade from 8.x [prepare-upgrade-from-8.x]
+
+To upgrade to 9.0 from 8.17 or earlier, you must first upgrade to the latest patch version of 8.18. This enables you to use the [Upgrade Assistant](prepare-to-upgrade/upgrade-assistant.md) to identify and resolve issues, reindex indices created before 8.0, and then perform a rolling upgrade. Upgrading to 8.18 before upgrading to 9.x is required even if you opt to do a full-cluster restart of your {{es}} cluster. If you're running a pre-8.x version, you might need to perform multiple upgrades or a full-cluster restart to get to 8.18 to prepare to upgrade to 9.0.
```

Comment:
```suggestion
To upgrade to 9.0 from 8.17 or earlier, you must first upgrade to the latest patch version of 8.18. This enables you to use the [Upgrade Assistant](prepare-to-upgrade/upgrade-assistant.md) to identify and resolve issues, reindex indices created before 8.0, and then perform a rolling upgrade. Upgrading to 8.18 before upgrading to 9.0 is required even if you opt to do a full-cluster restart of your {{es}} cluster. If you're running a pre-8.x version, you might need to perform multiple upgrades or a full-cluster restart to get to 8.18 to prepare to upgrade to 9.0.
```
Since 8.18 is only for 9.0, we can be specific.
When 9.1 comes out, it will be 8.19 to 9.1+